pathcore/network_permutation_test.py

```diff
@@ -0,0 +1,185 @@
+"""Methods for permuting the pathways across constructed features of one
+or more unsupervised models.
+A permutation maintains the number of unique pathways that correspond to a
+feature. If applicable, it also preserves the side (positive or negative)
+in which pathway was overrepresented as well.
+"""
+import csv
+
+from statsmodels.sandbox.stats.multicomp import multipletests
+
+
+def aggregate_permuted_network(observed_networks):
+    """This method handles the case where multiple observed networks are
+    generated (e.g. from models produced by different random seed
+    initializations of an unsupervised feature construction algorithm).
+    We handle analysis of multiple networks by aggregating them; likewise,
+    we require that the permutation test generates N aggregate permuted
+    networks.
+
+    Parameters
+    -----------
+    observed_networks : list(CoNetwork)
```

Comment:
Oh I didn't know that! Had only converted scripts to Python 3 recently - I will look into it! Might add "include type hints" as an issue on this & other relevant repos once everything has been checked in :+1: Thanks!